Xamarin.Forms.Core/Markup/ViewExtensions.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace Xamarin.Forms.Markup
+{
+	public static class ViewExtensions
+	{
```

Comment:
Modern fluent UI frameworks like **SwiftUI** (and @Clancey 's **Comet**) also use H and V instead of Horizontal and Vertical in choice places, e.g. they have `HStack` and `VStack`.

To me H and V makes the markup feel less noisy, quicker to scan.
Primary key words (nouns) stand out clearly; they are not obscured by long secondary adjectives.

E.g. compare:

`.CenterH ()` 

`.CenterHorizontal ()`

`Center` is primary here while `Horizontal` is secondary; the long form adds noise but no clarity.
Note that this is all about reading (which is 90% of time versus 10% writing).

To demonstrate the impact, I renamed all helpers in a large production app to the proposed names. Here are some real-world examples:

H and V:
```CSharp
new Label { Text = "Classify Call", LineBreakMode = LineBreakMode.TailTruncation } .Style (Styles.TitleAlone)
           .CenterExpandH () .TextCenterH () .CenterV ()

new Label { TextColor = Colors.c_FFFFFF_White } .Font (31)
            .FillExpandH () .CenterV () .TextCenterH ()

new Label { LineBreakMode = LineBreakMode.WordWrap } .Style (Styles.Header)
           .Row (Row.Prompt) .CenterExpandH () .TextCenterH ()

new Label { LineBreakMode = LineBreakMode.WordWrap } .Style (Styles.Header)
           .Row (Row.Message) .CenterExpandH () .Top () .Margins (bottom: 34) .TextCenterH ()
```
Horizontal and Vertical:
```CSharp
new Label { Text = "Classify Call", LineBreakMode = LineBreakMode.TailTruncation } .Style (Styles.TitleAlone)
           .CenterExpandHorizontal () .TextCenterHorizontal () .CenterVertical ()

new Label { TextColor = Colors.c_FFFFFF_White } .Font (31)
           .FillExpandHorizontal () .CenterVertical () .TextCenterHorizontal ()

new Label { LineBreakMode = LineBreakMode.WordWrap } .Style (Styles.Header)
           .Row (Row.Prompt) .CenterExpandHorizontal () .TextCenterHorizontal ()

new Label { LineBreakMode = LineBreakMode.WordWrap } .Style (Styles.Header)
           .Row (Row.Message) .CenterExpandHorizontal () .Top () .Margins (bottom: 34) .TextCenterHorizontal ()
```
For me it takes noticeably more effort to **scan the layout lines** (see markup conventions under Layout in the PR description) in the long version. Not a dramatic difference, but still short feels easier on the eyes.
So I am inclined to keep it short and clear.

To verify I will send out a Twitter poll for existing and interested devs.
Let's see if there is a clear majority preference and then decide.

_More background:
Markup is different from logic in that developers need to form a mental image of the location of many controls at once, by scanning (often quite large) chunks of source in one go. This is made easier by concise and consistent naming. We use a **layout line convention**._

_Common, relatively verbose naming guidelines were designed for procedural logic, not for declarative markup. [Modern insights in designing fluent API's](https://twitter.com/vincenth_net/status/1219695514276573184) also support removing noise:_
> _Don't name fluent API methods for clarity in isolation, but for clarity in their intended usage context_

_Also see under [CodeGen API would have little value above](https://github.com/xamarin/Xamarin.Forms/pull/8342#issuecomment-554350423):_
> _C# markup is improved by intelligently choosing shorter names for the most often used (combinations of) overly verbose property setters_

Reply:
To me, the second version scans better. The abbreviated version just looks like a bunch of typos at first glance. It's subjective.

Given that it's subjective, we should aim for consistency. General C#/.NET naming guidelines suggest using full words rather than abbreviations, and that's the guideline we follow in the rest of Xamarin.Forms [1]. Keeping the terms consistent throughout the API makes them more discoverable - if I type "horizontal" into a search or Intellisense, "CenterHorizontal" will show up while "CenterH" will not. 

If folks want to find the API to deal with columns, they should be able to trust that we call them "column" everywhere; not "column" in some places and "col" in others. 

[1] We've probably gotten at least one property name wrong somewhere, but it's _mostly_ consistent.

Reply:
Rather than subjective, I suspect the difference in preference has to do with whether someone is new to this API or has used it a lot.

However that only confirms the valid point that you raise about discoverability.

For the majority of devs this API is new; my [Twitter poll](https://twitter.com/vincenth_net/status/1226923217190825984) corroborates the preference:
![image](https://user-images.githubusercontent.com/1872271/74253562-845fff80-4cef-11ea-8ae2-151aa46429d8.png)

So to support the majority of devs better I renamed H -> Horizontal and V -> Vertical throughout.

Should developers who have gained experience with this API find that their preference changes, they can very easily add their own naming (and other) optimizations (potentially all the way to a tailor-made markup DSL, as @rdavisau [demonstrates](https://github.com/davidortinau/Xappy/blob/master/Xappy/Xappy/Content/Scenarios/Login/LoginPage.cs) in @davidortinau 's Xappy).